### ISSUE TITLE
Remove all uses of distutils

### ISSUE THREAD

**Issue by cool-RR** [tensorflow/tensorflow#58073]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.10.0

### Custom Code

No

### OS Platform and Distribution

N/A

### Python version

3.10

### Current Behaviour?

```shell
Tensorflow uses distutils, which will be removed in Python 3.12. This currently causes a `DeprecationWarning` to be printed in Python 3.10.
```

### Standalone code to reproduce the issue

```shell
import tensorflow
```

### Relevant log output

```shell
/home/ramrachum/.venvs/ray_env/lib/python3.10/site-packages/tensorflow/python/compiler/tensorrt/utils.py:21: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12
. Use setuptools or check PEP 632 for potential alternatives
```
</details>

**Comment by sachinprasadhs** [comment]
Looks like the issue has been addressed in the PR https://github.com/tensorflow/tensorflow/pull/55804 to cleanup distutils.
Could you please verify and close the issue. Thanks!

**Comment by mihaimaruseac** [comment]
The timing of this issue, the version number, the timing of the PR, clearly state that this is not the case.

Plus, there are several places where `distutils` is still mentioned: https://cs.opensource.google/search?q=distutils&sq=&ss=tensorflow%2Ftensorflow:tensorflow%2F

**Comment by loqs** [comment]
Arch linux encountered this with the 2.16.1 release when it switch to Python 3.12 which removed the distutils module.
The two instances of distutils made it in the packaged tensorflow were https://github.com/tensorflow/tensorflow/blob/v2.16.1/tensorflow/api_template.__init__.py#L30 and https://github.com/tensorflow/tensorflow/blob/v2.16.1/tensorflow/lite/python/convert.py#L17

**Comment by loqs** [comment]
@pyrox0 would a patch such as [tensorflow-2.16.1-python-distutils-removal.patch](https://gitlab.archlinux.org/archlinux/packaging/packages/tensorflow/-/blob/fff04b07ca2bc6758f81505cc396c91f7466c08a/tensorflow-2.16.1-python-distutils-removal.patch) or having setuptools which vendors distutils installed be suitable options for NixOS?

**Comment by apraga** [comment]
@angerson Hi, would it be possible to apply the archlinux patch mentioned above ? It seems to fix the issue. Thanks !

**Comment by GaetanLepage** [comment]
> @angerson Hi, would it be possible to apply the archlinux patch mentioned above ? It seems to fix the issue. Thanks !

This would help us at [nixpkgs](https://github.com/NixOS/nixpkgs) as well :)

**Comment by drupol** [comment]
> This would help us at [nixpkgs](https://github.com/NixOS/nixpkgs) as well :)

Can't we just re-use the patch in `nixpkgs` ?

**Comment by mihaimaruseac** [comment]
Can you please submit the patch as a PR? I'll try to review and shepherd it in the internal pipeline.

**Comment by drupol** [comment]
Sure, here you go: https://github.com/tensorflow/tensorflow/pull/70959

**Comment by drupol** [comment]
Apparently, that patch was not enough: https://github.com/NixOS/nixpkgs/pull/325224#issuecomment-2221678874

**Comment by apraga** [comment]
This should be closed with #74090 .

**Comment by mihaimaruseac** [comment]
That is correct. Thank you!
